internal/impl/kafka/enterprise/redpanda_migrator_output.go

```diff
@@ -220,156 +158,36 @@ func NewRedpandaMigratorWriterFromConfig(conf *service.ParsedConfig, mgr *servic
-	if err != nil {
-		return nil, err
-	}
-	for _, b := range brokerList {
-		w.SeedBrokers = append(w.SeedBrokers, strings.Split(b, ",")...)
-	}
-
-	if w.topic, err = conf.FieldInterpolatedString("topic"); err != nil {
-		return nil, err
-	}
-
-	if conf.Contains("key") {
-		if w.key, err = conf.FieldInterpolatedString("key"); err != nil {
-			return nil, err
-		}
-	}
-
-	if conf.Contains("partition") {
-		if rawStr, _ := conf.FieldString("partition"); rawStr != "" {
-			if w.partition, err = conf.FieldInterpolatedString("partition"); err != nil {
-				return nil, err
-			}
-		}
-	}
-
-	if w.timeout, err = conf.FieldDuration("timeout"); err != nil {
-		return nil, err
-	}
-
-	maxMessageBytesStr, err := conf.FieldString("max_message_bytes")
-	if err != nil {
-		return nil, err
-	}
-	maxMessageBytes, err := humanize.ParseBytes(maxMessageBytesStr)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse max_message_bytes: %w", err)
-	}
-	if maxMessageBytes > uint64(math.MaxInt32) {
-		return nil, fmt.Errorf("invalid max_message_bytes, must not exceed %v", math.MaxInt32)
-	}
-	w.produceMaxBytes = int32(maxMessageBytes)
-	brokerWriteMaxBytesStr, err := conf.FieldString("broker_write_max_bytes")
-	if err != nil {
-		return nil, err
-	}
-	brokerWriteMaxBytes, err := humanize.ParseBytes(brokerWriteMaxBytesStr)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse broker_write_max_bytes: %w", err)
-	}
-	if brokerWriteMaxBytes > 1<<30 {
-		return nil, fmt.Errorf("invalid broker_write_max_bytes, must not exceed %v", 1<<30)
-	}
-	w.brokerWriteMaxBytes = int32(brokerWriteMaxBytes)
-
-	if conf.Contains("compression") {
-		cStr, err := conf.FieldString("compression")
-		if err != nil {
-			return nil, err
-		}
-
-		var c kgo.CompressionCodec
-		switch cStr {
-		case "lz4":
-			c = kgo.Lz4Compression()
-		case "gzip":
-			c = kgo.GzipCompression()
-		case "snappy":
-			c = kgo.SnappyCompression()
-		case "zstd":
-			c = kgo.ZstdCompression()
-		case "none":
-			c = kgo.NoCompression()
-		default:
-			return nil, fmt.Errorf("compression codec %v not recognised", cStr)
-		}
-		w.compressionPrefs = append(w.compressionPrefs, c)
-	}
-
-	w.partitioner = kgo.StickyKeyPartitioner(nil)
-	if conf.Contains("partitioner") {
-		partStr, err := conf.FieldString("partitioner")
-		if err != nil {
-			return nil, err
-		}
-		switch partStr {
-		case "murmur2_hash":
-			w.partitioner = kgo.StickyKeyPartitioner(nil)
-		case "round_robin":
-			w.partitioner = kgo.RoundRobinPartitioner()
-		case "least_backup":
-			w.partitioner = kgo.LeastBackupPartitioner()
-		case "manual":
-			w.partitioner = kgo.ManualPartitioner()
-		default:
-			return nil, fmt.Errorf("unknown partitioner: %v", partStr)
-		}
-	}
-
-	if w.clientID, err = conf.FieldString("client_id"); err != nil {
-		return nil, err
-	}
-
-	if w.rackID, err = conf.FieldString("rack_id"); err != nil {
+	var err error
+	if w.recordConverter, err = kafka.NewFranzWriterFromConfig(conf, nil, nil); err != nil {
```

Comment:
The docs for `NewFranzWriterFromConfig` say:
> A closure function must be provided that is responsible for granting access to a connected client.

Might be worth updating that to mention it's optional.